src/Infrastructure.EntityFramework/AdminConsole/Repositories/OrganizationRepository.cs

```diff
@@ -270,8 +271,199 @@ into grouped
         return await query.ToListAsync();
     }
 
-    public Task EnableCollectionEnhancements(Guid organizationId)
+    public async Task EnableCollectionEnhancements(Guid organizationId)
     {
-        throw new NotImplementedException("Collection enhancements migration is not yet supported for Entity Framework.");
+        using var scope = ServiceScopeFactory.CreateScope();
+        var dbContext = GetDatabaseContext(scope);
+        await using var transaction = await dbContext.Database.BeginTransactionAsync();
+
+        try
+        {
+            // Step 1: AccessAll migration for Groups
+            var groupIdsWithAccessAll = dbContext.Groups
+                .Where(g => g.OrganizationId == organizationId && g.AccessAll == true)
+                .Select(g => g.Id)
+                .ToList();
+
+            // Update existing CollectionGroup rows
+            dbContext.CollectionGroups
+                .Where(cg =>
+                    groupIdsWithAccessAll.Contains(cg.GroupId) &&
+                    cg.Collection.OrganizationId == organizationId)
+                .ToList()
+                .ForEach(cg =>
+                {
+                    cg.ReadOnly = false;
+                    cg.HidePasswords = false;
+                    cg.Manage = false;
+                });
+
+            // Insert new rows into CollectionGroup
+            foreach (var group in groupIdsWithAccessAll)
+            {
+                var newCollectionGroups = dbContext.Collections
+                    .Where(c =>
+                        c.OrganizationId == organizationId &&
+                        !dbContext.CollectionGroups.Any(cg => cg.CollectionId == c.Id && cg.GroupId == group))
+                    .Select(c => new CollectionGroup
+                    {
+                        CollectionId = c.Id,
+                        GroupId = group,
+                        ReadOnly = false,
+                        HidePasswords = false,
+                        Manage = false
+                    })
+                    .ToList();
+                dbContext.CollectionGroups.AddRange(newCollectionGroups);
+            }
+
+            // Update Group to clear AccessAll flag
+            dbContext.Groups
+                .Where(g => groupIdsWithAccessAll.Contains(g.Id))
+                .ToList()
+                .ForEach(g => g.AccessAll = false);
+
+            // Save changes for Step 1
+            await dbContext.SaveChangesAsync();
```

Comment:
Please double check how this works, but I suspect that `dbContext.SaveChangesAsync` will work against your transaction commit/rollback statements. That is, it'll write your changes to the db so you can no longer roll them back - I think you can only use one or the other. Is there any need to commit our work partway through?

Reply:
The data is only actually persisted after we commit the transaction (see here: https://learn.microsoft.com/en-us/ef/core/saving/transactions#controlling-transactions).


> Is there any need to commit our work partway through?

Yes to retrieve the dirty data on subsequent queries.

Reply:
Thanks for the link, makes sense to me.